olive/engine/engine.py

```diff
@@ -406,6 +374,65 @@ def run(
 
         return outputs
 
+    def run_accelerator(
+        self,
+        input_model_config: ModelConfig,
+        data_root: str,
+        output_dir: str,
+        output_name: str,
+        evaluate_input_model: bool,
+        accelerator_spec: AcceleratorSpec,
+    ):
+        # generate search space and initialize the passes for each hardware accelerator
+        self.setup_passes(accelerator_spec)
+
+        # hash the input model
+        input_model_id = self._init_input_model(input_model_config)
+        self.footprints[accelerator_spec].record(model_id=input_model_id)
+
+        try:
+            if evaluate_input_model:
+                prefix_output_name = (
+                    f"{output_name}_{accelerator_spec}_" if output_name is not None else f"{accelerator_spec}"
+                )
+                assert self.evaluator_config is not None, "evaluate_input_model is True but no evaluator provided"
+                results = self._evaluate_model(
+                    input_model_config, input_model_id, data_root, self.evaluator_config, accelerator_spec
+                )
+                logger.info(f"Input model evaluation results: {results}")
+                result_name = f"{prefix_output_name}_input_model_metrics"
+                results_path = output_dir / f"{result_name}.json"
+                with open(results_path, "w") as f:
+                    json.dump(results.to_json(), f, indent=4)
+                logger.info(f"Saved evaluation results of input model to {results_path}")
+                if not self.passes:
+                    logger.debug("No passes registered, return input model evaluation results.")
```

Comment:
For the case you mentioned, when there is no passes, the input model only be evaluated once. It should be a bug.
How to extract the logics of input model evaluation function out of `run_accelerator?`

Reply:
Actually, I am looking forward to unifying them. However, from the current footprint implementation, it seems a little bit hard to unify them. 

How could we do it?

Reply:
Basically, the run_accelerator is invoked under multiple EP context manager. If the evaluation model doesn't support multiple EP, it is ok to move it out of run_accelerator. Otherwise, we need make all of these logic under the context manager.

Reply:
Agree. The evaluation is bound with accelerate spec. Making all these logics under the same context is OK for me.

For unify the results, will do it in following PR.